### PR TITLE
Add a few worakrounds for problematic r functions

### PR DESCRIPTION
**Pull request by janschulz**
* Workaround a problem with vignette(): closes: https://github.com/IRkernel/IRkernel/issues/267
* we have no `edit()`and `View()`: https://github.com/IRkernel/IRkernel/issues/280

### CODE REVIEW

File: R/environment.r [hunk 1]
```diff
@@ -17,4 +17,26 @@ init_shadowenv <- function() {
     add_to_user_searchpath('.irk.get_shadowenv', get_shadowenv)
     add_to_user_searchpath('.irk.add_to_user_searchpath', add_to_user_searchpath)
 
+    # For the rest of the functions, please explain why the workaround is needed
+    # (=the problem) and link to the issue describing the problem.
+
+    # workaround for problems with vignette(xxx) not bringing up the vignette
+    # content in the browser: https://github.com/IRkernel/IRkernel/issues/267
+    add_to_user_searchpath('print.vignette', function(...){
+        utils:::print.vignette(...)
```
Comment (flying-sheep) [inline R/environment.r]: ::: means private access. is this a private function?

Comment (janschulz) [inline R/environment.r]:
```R
> utils::print.vignette
Error: 'print.vignette' is not an exported object from 'namespace:utils'
> 
```

Comment (takluyver) [inline R/environment.r]: Is that going to be a problem for the checks CRAN does?

Comment (flying-sheep) [inline R/environment.r]: yes, that’s why i mentioned it.

but we can use [this hack](https://github.com/IRkernel/repr/blob/df33b1574c8abb5eb7c19e7cc33fafbd60847e13/R/repr_help_files_with_topic.r#L12)

Comment (takluyver) [inline R/environment.r]: Get your package on Cran with this one weird trick! ;-)

Comment (janschulz) [inline R/environment.r]: done

**Comment by janschulz** [comment]
Anything else?

**Comment by janschulz** [comment]
https://github.com/IRkernel/IRkernel/issues/85
-> implement getting raw input from the frontend... But that's IMO for a different PR because it touches the messaging system.

So IMO this is ready for review and eventual merging...

File: R/environment.r [hunk 2]
```diff
@@ -11,10 +11,36 @@ get_shadowenv <- function() {
...
     # from everywhere...
     add_to_user_searchpath('.irk.get_shadowenv', get_shadowenv)
     add_to_user_searchpath('.irk.add_to_user_searchpath', add_to_user_searchpath)
 
+    # For the rest of the functions, please explain why the workaround is needed
+    # (=the problem) and link to the issue describing the problem.
+
+    # workaround for problems with vignette(xxx) not bringing up the vignette
+    # content in the browser: https://github.com/IRkernel/IRkernel/issues/267
+    add_to_user_searchpath('print.vignette', function(...){
```
Comment (flying-sheep) [inline R/environment.r]: `){` → `) {`

File: R/environment.r [hunk 3]
```diff
@@ -11,10 +11,36 @@ get_shadowenv <- function() {
...
+        # returning immediately will run into trouble with zmq and its polling
+        # preventing the vignette server to startup. So wait a little to let
+        # it startup...
+        # 0.1 is too little, so add some margin...
+        Sys.sleep(0.5)
+    })
+
+    # we simply have currently no way to view or edit dfs:
+    # https://github.com/IRkernel/IRkernel/issues/280
+    add_to_user_searchpath('View', function(...){
```
Comment (flying-sheep) [inline R/environment.r]: `){` → `) {`

File: R/environment.r [hunk 4]
```diff
@@ -11,10 +11,36 @@ get_shadowenv <- function() {
...
+        stop("'View()' not yet supported in the Jupyter R kernel")
+    })
+    add_to_user_searchpath('edit', function(...){
```
Comment (flying-sheep) [inline R/environment.r]: `){` → `) {`

**Comment by flying-sheep** [comment]
looks great! have you confirmed that they works? then i’d just merge this…

File: R/environment.r (hunk 3 above)
Comment (takluyver) [inline R/environment.r]: :expressionless:

File: R/environment.r (hunk 2 above)
Comment (janschulz) [inline R/environment.r]: I switched to rstudio to lint the code and it doesn't show me this. Do we really want to waste time and effort on something which the RStudio inter doesn not catch?

Comment (janschulz) [inline R/environment.r]: done BTW...

File: R/environment.r (hunk 3 above)
Comment (janschulz) [inline R/environment.r]: done

File: R/environment.r (hunk 4 above)
Comment (janschulz) [inline R/environment.r]: done

**Comment by janschulz** [comment]
both work:

```
>>> View(data.frame())
Error in View(data.frame()): 'View()' not yet supported in the Jupyter R kernel
Traceback:

1. View(data.frame())
2. stop("'View()' not yet supported in the Jupyter R kernel")

>>> vignette("knitr")
starting httpd help server ... done
# and the vignette is displayed
```

**Pull request merged**